src/lib.rs

```diff
@@ -395,6 +425,9 @@ impl Client {
                     }
                 } => {}
             }
+
+            // Might fail if all Clients have been dropped, which means nobody is waiting for shutdown
+            let _ = wait_for_shutdown_tx.send(());
```

Comment:
Since both this heartbeat task, and the faktory connection tasks have send sides of this channel, and (I believe) that we want the `self.wait_for_shutdown_rx.recv().await` in `close()` to block until both of them have exited, shouldn't we not `send` either here, or in `spawn_faktory_connection`, and instead rely on the tx side being dropped when those tasks finish to cause the `recv()` to stop blocking?

Reply:
Yeah I think we can rely on dropping, but it didn't seem to hurt to just send to it, spawn_faktory_connection will only send if the heartbeat task isn't there (so a producer)

Reply:
Not strictly a problem, just easy to forget (and lose track of) which thing is the "last" one to exit, and that only the last one should be sending at all. Establishing the drop pattern (and maybe commenting that this is explicitly what we're doing) lessens the chances of accidentally sending where/when it shouldn't in the future, even though those are the only two places right now.

Reply:
You are totally right, will do that!

Reply:
Done!